packages/esm-bed-management-app/src/summary/summary.resource.ts

```diff
@@ -0,0 +1,185 @@
+import useSWR from 'swr';
+import { type FetchResponse, openmrsFetch } from '@openmrs/esm-framework';
+import type { AdmissionLocation, Bed, MappedBedData } from '../types';
+
+export const useLocationsByTag = (locationUuid: string) => {
+  const locationsUrl = `/ws/rest/v1/location?tag=${locationUuid}&v=full`;
+
+  const { data, error, isLoading, isValidating, mutate } = useSWR<{ data }, Error>(
+    locationUuid ? locationsUrl : null,
+    openmrsFetch,
+  );
+
+  return {
+    data: data?.data?.results ?? [],
+    error,
+    isLoading,
+    isValidating,
+    mutate,
+  };
+};
+
+export const getBedsForLocation = (locationUuid: string) => {
```

Comment:
I sort of understand having a `use` hook and a `get` function for the same thing (though, not really), but `getBedsForLocation` and `findBedByLocation` are basically duplicative.

Reply:
`getBedsForLocation` is cruft. `findBedByLocation` is called many times by the [same component](https://github.com/openmrs/openmrs-esm-patient-management/blob/74a79e963fa561a900d3cc0d6616760e09a6bd33/packages/esm-bed-management-app/src/bed-administration/bed-administration-table.component.tsx#L87-L108) to fetch beds for an array of locations. This can probably be simplified (or all the logic moved into a hook, at least).

Reply:
That's definitely something that should go in the "improve the API" queue.